haigha/transports/socket_transport.py

```diff
@@ -29,26 +34,33 @@ def connect(self, (host, port), klass=socket.socket):
         Connect assuming a host and port tuple.
         '''
         self._host = "%s:%s" % (host, port)
-        self._sock = klass()
-        self._sock.setblocking(True)
-        self._sock.settimeout(self.connection._connect_timeout)
-        if self.connection._sock_opts:
-            for k, v in self.connection._sock_opts.iteritems():
-                family, type = k
-                self._sock.setsockopt(family, type, v)
-        infos = socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP)
-        exc = socket.error("getaddrinfo returns an empty list")
-        for _family, _socktype, _proto, _canonname, sockaddr in infos:
+
+        for info in socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP):
+
+            family, socktype, proto, _, sockaddr = info
+            self._sock = klass(family, socktype, proto)
+            self._sock.setblocking(True)
+            self._sock.settimeout(self.connection._connect_timeout)
+            if self.connection._sock_opts:
+                for k, v in self.connection._sock_opts.iteritems():
+                    family, type = k
+                    self._sock.setsockopt(family, type, v)
             try:
+
                 self._sock.connect(sockaddr)
-                break
-            except socket.error as exc:
+
+            except Exception:
```

Comment:
`Exception` is probably too general. Use `socket.error` specifically. Using `Exception` could masquerade bugs in our own code (e.g., AttributeError, ValueError, TypeError, etc.). `socket.error` is the base class for the other socket exceptions. See "echo client program" example in https://docs.python.org/2/library/socket.html